packages/customize-widgets/README.md

```diff
@@ -14,4 +14,16 @@ npm install @wordpress/customize-widgets
 
 _This package assumes that your code will run in an **ES2015+** environment. If you're using an environment that has limited or no support for ES2015+ such as IE browsers then using [core-js](https://github.com/zloirock/core-js) will add polyfills for these methods._
 
+## Technical implementation details
+
+The new Widgets Customizer replaces `Appearance > Customize > Widgets` with block editors. The original Customizer is a Backbone app, but the editor is a React app. One of the challenges is to integrate them together but make sure features from both sides still work.
+
+We extend Customizer's sections and controls in the `controls` directory and inject some custom logic for the editor. We use React portal to render each editor in its section to reuse most of the styles and scripts provided by the Customizer.
+
+`components/sidebar-block-editor` is the entry point for each widget area's block editor. `component/sidebar-block-editor/sidebar-adapter.js` is an adapter to talk to the customize API and transform widget objects into widget instances. `components/sidebar-block-editor/use-sidebar-block-editor.js` is a custom React Hook to integrate the adapter into React and handle most of the translations between blocks and widgets. These allow us to implement basic editing features as well as real-time preview in a backward-compatible way.
+
+Whenever the blocks change, we run through each block to determine if there are created, edited, or deleted blocks. Then, we converted them to their widget counterparts and call the customize API to update them. For React developers, it can be seen as a custom reconciler or a custom renderer for the Customizer. But instead of targeting DOM as the render target, we are targeting WordPress widgets using the customize API.
```

Comment:
```suggestion
Whenever the blocks change, we run through each block to determine if there are created, edited, or deleted blocks. We then convert them to their widget counterparts and call the Customize API to update them. 

For React developers, this can be thought of as a custom reconciler or a custom renderer for the Customizer. But instead of targeting DOM as the render target, we are targeting WordPress widgets using the Customize API.
```